config/resources/test_config_null_nested_fields.json

```diff
@@ -5,11 +5,23 @@
     "enabled": null,
     "method": null,
     "includeEntity": null,
+    "datacenter": null,
     "meta": null,
     "region": null,
     "stsEndpoint": null,
     "serverIdHeaderValue": null
   },
+  "consulServers": {
+    "hosts": "",
+    "https": null,
+    "httpPort": null,
+    "grpcPort": null,
+    "httpsCACertFile": null,
+    "caCertFile": null,
+    "tls": null,
+    "tlsServerName": null,
+    "skipServerWatch": null
```

Comment:
> I think consul-k8s assumes the same TLS settings for https and grpc+tls to the Consul servers?

We support configuring individual TLS CA certs for HTTP and gRPC in the current version of Consul ECS. The mesh task's terraform module exposes two variables to control that. So I guess we should continue maintaining both the settings going forward as well. What do you think?

That said, I do like the way you restructured these inputs so that users can go with a default setting for both the protocols. I will merge this PR in (since it has already become huge) and raise a separate one to address this restructuring.